pyanaconda/modules/storage/partitioning/manual/manual_partitioning.py

```diff
@@ -91,6 +91,26 @@ def _setup_mount_point(self, storage, mount_data):
 
             if device.raw_device.type in ("btrfs volume", "btrfs subvolume"):
                 # 'Format', or rather clear the device by recreating it
+
+                # recreating @device will remove all nested subvolumes of it, we cannot allow
+                # using these nested subvolumes for other MountPointRequest without also
+                # re-creating them
+                if device.raw_device.type == "btrfs volume":
+                    depending_subvolumes = device.raw_device.subvolumes
+                elif device.raw_device.type == "btrfs subvolume":
+                    depending_subvolumes = [sub.name for sub in device.raw_device.volume.subvolumes
```

Comment:
Here you check only for one nest level right? What about some children deep in the tree?

Reply:
Aha so great - would be nice to expose this property in the DeviceData for the front end to use instead of the recursive function I did.

Reply:
It's not perfect, because you need some extra steps to make sure to get "nested subvolumes for a subvolume" so we'd need to expose the `depends_on` function as well and I am not sure that would make things that simpler in the frontend. I think we need some property like `descendats` in Blivet that would be exposed in the API (we already have the `ancestors` property going the other way).